File: sdk/python/hitechcloud/resources/proxmox_backup.py
"""Proxmox Backup resource (9 endpoints)"""

from .base import BaseResource

class ProxmoxBackupResource(BaseResource):
    """Proxmox Backup: PBS connection, credentials, usage, snapshots, groups, password, token"""

    def get_connection_info(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs - Get backup connection info"""
        return self._http.get(f"/api/service/{service_id}/pbs")

    def get_credentials(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs/credentials - Get backup credentials"""
        return self._http.get(f"/api/service/{service_id}/pbs/credentials")

    def get_usage(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs/usage - Get backup usage"""
        return self._http.get(f"/api/service/{service_id}/pbs/usage")

    def get_metrics(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs/metrics - Get backup metrics"""
        return self._http.get(f"/api/service/{service_id}/pbs/metrics")

    def list_snapshots(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs/snapshots - List snapshots"""
        return self._http.get(f"/api/service/{service_id}/pbs/snapshots")

    def list_groups(self, service_id: int) -> dict:
        """GET /api/service/{id}/pbs/groups - List backup groups"""
        return self._http.get(f"/api/service/{service_id}/pbs/groups")

    def change_password(self, service_id: int, password: str) -> dict:
        """POST /api/service/{id}/pbs/password - Change backup password"""
        return self._http.post(f"/api/service/{service_id}/pbs/password", {"password": password})

    def rotate_token(self, service_id: int) -> dict:
        """POST /api/service/{id}/pbs/token - Rotate API token"""
        return self._http.post(f"/api/service/{service_id}/pbs/token")

    def revoke_token(self, service_id: int) -> dict:
        """DELETE /api/service/{id}/pbs/token - Revoke API token"""
        return self._http.delete(f"/api/service/{service_id}/pbs/token")

File: sdk/ruby/lib/hitechcloud/resources/proxmox_backup.rb
module HiTechCloud
  # ProxmoxBackup resource
  class ProxmoxBackupResource < BaseResource

    # Get PBS info
    def pbs(id)
      @http.get("/api/service/%{id}/pbs" % { id: id })
    end

    # Get credentials
    def credentials(id)
      @http.get("/api/service/%{id}/pbs/credentials" % { id: id })
    end

    # Get usage
    def usage(id)
      @http.get("/api/service/%{id}/pbs/usage" % { id: id })
    end

    # Get metrics
    def metrics(id)
      @http.get("/api/service/%{id}/pbs/metrics" % { id: id })
    end

    # Get snapshots
    def snapshots(id)
      @http.get("/api/service/%{id}/pbs/snapshots" % { id: id })
    end

    # Get groups
    def groups(id)
      @http.get("/api/service/%{id}/pbs/groups" % { id: id })
    end

    # Change password
    def change_password(id, data)
      @http.post("/api/service/%{id}/pbs/password" % { id: id }, data)
    end

    # Create token
    def create_token(id, data)
      @http.post("/api/service/%{id}/pbs/token" % { id: id }, data)
    end

    # Delete token
    def delete_token(id)
      @http.delete("/api/service/%{id}/pbs/token" % { id: id })
    end

  end
end
